### ISSUE TITLE
TR4400 v2 / RAC2V1A

### ISSUE THREAD

**Issue by 4q1** [Lanchon/openwrt-tr4400-v2#2]
I am curious if this also applies to the SAC2V1A.  I found one of these in the trash - It's branded Spectrum (thus the "S" in the model?)

The internals look the same.  I was able to setup a serial connection, but never get a root (or any) prompt.  At the default bit rate of 115200, I do see characters on the terminal screen.  If I change the baud, sometime the correct characters are printed and sometimes not. I assume that means the serial connection is working and the characters are making the round trip to/from the router. Correct?

When the Router boots, it flashes blue for ~30 seconds, then slowly fades blue in and out at 1 - 2 hertz.  then after a minute or two it fades red in and out at the same frequency.  I am wonder if the problem is that it never completely boots.  I've tried connecting to the web interface, but it says <the equivalent of the certs being out of date> and will not let me connect. It does auto redirect from http to https. I plugged it into my spectrum modem in the hopes of it getting updates, but nothing new happens.  Same colored light pattern.  

Now What?

**Comment by Lanchon** [comment]
it completed booting alright.

first lets identify that router.

take a look here: https://www.ebay.ca/itm/225664504213

note that the model number RAC2V1A is here, bottom left beside of the power jack:
![image](https://github.com/Lanchon/openwrt-tr4400-v2/assets/3977813/bd8c5ff7-152c-4fc5-829c-c399ac5238df)

and not in the label here:
![image](https://github.com/Lanchon/openwrt-tr4400-v2/assets/3977813/87039eeb-2402-4e95-ae4e-d92f68df599c)

the SAC2V1A is the software. AFAICT it is the same hardware but with a different software. SAC2V1A requires setting up online via QR code and an app... trash.

take a look here, this spectrum router (different manufacturer) also has the two software variants, so im extrapolating from that: https://openwrt.org/inbox/toh/askey/rt4230w_rev6

so your device, is it a RAC2V1A as shown at the bottom by the power jack?

if it is, i guess your serial connection is not working. it cannot "sometimes" show text and not other times. in order for you to get a shell you need to press enter when it finishes booting. to press enter you need a working 1.8V serial adapter. if you used 3.3V you probably already fried the port.

also, since this is a different software, it might deny the shell (but i doubt it).

i really wish there was a way to hack these devices and escalate privilege. i tried looking for an exploitable bug on mine (of the known bugs openwrt had), but i couldn't find any. (this search was after i had dumped the firmware, of course).

**Comment by 4q1** [comment]
Thanks for the pictures!  you are correct, I was looking at the label on the top, not the printing on the bottom.
It is indeed a RAC2V1A.

I have to question you about the serial port not showing text sometimes. Over the years, decreasingly, I've had to use terminal programs.  Often there were problems with double characters because the terminal would print the characters typed and the host end would send those back as well.  In this case, the terminal doesn't seem to be printing the characters, but the router is sending them back.  At the right baud everything works great, and the wrong baud there are mistakes, and at a very wrong baud it's either gibberish or nothing at all. I am fairly certain this means the serial connection is working... It **_is_** getting characters and sending them back correctly.

Thanks for great write-up BTW.  I could have saved myself a lot by just buying a (better?)  EA6350 on ebay, but I was up for a challenge yesterday.  I found an old USB<=>Serial connector, two 10K and one 470ohm resistors.  I only bridged the bottom two tiny connections (r60 & r61 you highlighted in blue), and used the two S pads to solder on a jumper leading out to the Rx & Tx resistors. I did use the grounding hole though. This was easier for a temporary setup.  I just left the two boards sit out on the table while working on it.  not ideal, but works temporary - or should have in theory. 

googling for the LED patterns I can't find any slow-pulsating red descriptions.  that's why I was wondering if it was booting at all.

**Comment by Lanchon** [comment]
> I can't find any slow-pulsating red descriptions

this is when the device cannot phone home. wire the wan port to the internet and it should day breathing in blue.

could you please capture the complete boot sequence, including noise, and post it as a gist or email it to me? (my nick here on github at gmail)

**Comment by 4q1** [comment]
the manual differentiates between pulsing and flashing.  It's not the flashing like the initial blue flash when it boots.  it's the fade in and out red. I have connected it to the network (wan port) and rebooted several times.  No joy.  

I will sent the light sequence. I need a few days to get back to it. other things have piled up.

**Comment by Lanchon** [comment]
> could you please capture the complete boot sequence, including noise, and post it as a gist or email it to me? (my nick here on github at gmail)

i dont care about the led. i want a complete text capture of the serial port (including noise and garbage) during the complete boot sequence from the moment you power it up.

**Comment by 4q1** [comment]
Ah, got it.  you would have been disappointed. This is all moot, but...

There was no bootup output, just what I typed. no other output from the router.  
If I were to type this at 115200, it would have work perfectly

> this is a test

but at 38400 it would have looked like

> phis is c pest

I was surprised it worked as good as it did. I'm confidant the serial connection was working at 115200 baud. However, nothing would show during the boot. Today I plugged it all in to try again, and I got a solid red light seconds after powering up. My heart dropped. After playing with it a while I saw that my ground wire came out. reattaching it didn't seem to make a difference. I must have shorted something momentarily - probably fried the UART.  Eventually I did get it back to the blue boot sequence, but after that I was not able to get anything out of the serial connect at all - even what I typed.  Thinking that some of those solder bridges may have been a via, I did bridge all the connections as shown in the tutorial and tried again. no luck.

I'm calling it a lost cause.  I should have taken more care soldering on the pigtail leading from the board to the limiting resistors.

Thanks for your time and super fast responses. When I find another I may try again.
